Automattic/woocommerce-payments
# Add Payment Request button support for Woo blocks

Fixes #1416
Fixes #1544
Fixes #1677

#### Changes proposed in this Pull Request

- Add Payment Request blocks integration. The code is adapted from the Stripe Payment Gateway (1467-gh-woocommerce/woocommerce-gateway-stripe).
- Use same normalization functions and API in blocks and shortcode.
- Remove some leftover code from shortcode integration (e.g. `clearCart`, unnecessary i18n and nonce variables) and make shortcode integration more similar to the classic checkout by using the `WCPayAPI`.
- Fix 3D Secure / SCA confirmation with payment request buttons.

There are a few improvements left to do:
- Perhaps change folder structure, moving everything related to blocks into `client/blocks/`. - This requires changing the current credit card checkout as well.
- Add loading indicator to PRB when 3DS modal is open or intent is being validated. - Not really related to this PR, as it applies to the shortcode implementation too.

#### Testing instructions

- Run `npm run build:client`.
- Install Gutenberg and WooCommerce Blocks.
- Create a second cart page and add the cart block to it.
- Create a second checkout page and add the checkout block to it.
- Test using the payment request button in the product, cart and checkout pages, both shortcode and blocks.
- Verify that the button shows up as expected.
- Verify that the order is created as expected when the payment succeeds.

-------------------
